### PR TITLE
[12.1] ISPN-13014 LDAP identity recursive search

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13014

### CODE REVIEW

File: server/runtime/src/main/resources/schema/infinispan-server-12.1.xsd [hunk 1]
```diff
@@ -1268,6 +1268,21 @@
...
+         <xs:annotation>
+            <xs:documentation>
+               Indicates if queries are recursive.
+            </xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="filter-name" type="xs:string" default="(rdn_identifier={0})">
+         <xs:annotation>
+            <xs:documentation>
+               The LDAP filter for getting identity by name.
```
Comment (oraNod) [inline server/runtime/src/main/resources/schema/infinispan-server-12.1.xsd:1281]:
```suggestion
               Specifies the LDAP filter that retrieves an identity by name.
```

File: server/runtime/src/main/resources/schema/infinispan-server-12.1.xsd [hunk 2]
```diff
@@ -1268,6 +1268,21 @@
...
+               The string "{0}" will be replaced by the searched identity name and the "rdn_identifier" will be the value of the "rdn-identifier" attribute.
```
Comment (oraNod) [inline server/runtime/src/main/resources/schema/infinispan-server-12.1.xsd:1282]:
```suggestion
               In the default value, "{0}" is replaced with the searched identity name and "rdn_identifier" is replaced with the value of the "rdn-identifier" attribute.
```

File: server/runtime/src/main/resources/schema/infinispan-server-12.1.xsd [hunk 3]
```diff
@@ -1268,6 +1268,21 @@
             </xs:documentation>
          </xs:annotation>
       </xs:attribute>
+      <xs:attribute name="search-recursive" type="xs:boolean" default="false">
+         <xs:annotation>
+            <xs:documentation>
+               Indicates if queries are recursive.
```
Comment (oraNod) [inline server/runtime/src/main/resources/schema/infinispan-server-12.1.xsd:1274]:
```suggestion
               Performs recursive queries.
```

**Review by oraNod — CHANGES_REQUESTED** [review]

**Comment by tristantarrant** [comment]
I've squashed all of @oraNod 's doc fixes

**Comment by danberindei** [comment]
Merged, thanks Tristan!

**Pull request merged**